### DISCUSSION TITLE
Confusion over prefetching

### DISCUSSION THREAD

**Discussion by calbro7** [thecodingmachine/graphqlite#429] [Q&A]
Hi, I'm a bit confused by the sample code in the prefetching section of the docs:

```
#[Type]
class PostType {
    /**
     * @param Post $post
     * @param mixed $prefetchedUsers
     * @return User
     */
    #[Field(prefetchMethod: "prefetchUsers")]
    public function getUser(Post $post, $prefetchedUsers): User
    {
        // This method will receive the $prefetchedUsers as second argument. This is the return value of the "prefetchUsers" method below.
        // Using this prefetched list, it should be easy to map it to the post
    }

    /**
     * @param Post[] $posts
     * @return mixed
     */
    public function prefetchUsers(iterable $posts)
    {
        // This function is called only once per GraphQL request
        // with the list of posts. You can fetch the list of users
        // associated with this posts in a single request,
        // for instance using a "IN" query in SQL or a multi-fetch
        // in your cache back-end.
    }
}
```

The class is PostType but all the type annotations are for Post. Is this a mistake and the types are supposed to match up? Or else what is Post?

**Comment by oojacoboo** [comment]
It doesn't seem like the greatest example.  I'm assuming it's intended to mean a relation to PostType.  Please feel free to submit a PR with doc updates if you can improve the example.

**Reply by calbro7** [reply]
Sorry I'm still confused. You're saying that Post and PostType are two different entities such that a Post has a PostType?

**Comment by TamasSzigeti** [comment]
Not sure how to make it clearer, maybe by mentioning that it is using [external type mapping](https://graphqlite.thecodingmachine.io/docs/external-type-declaration)

**Reply by oojacoboo** [reply]
Well, we could make it more real-world applicable.  How does a `PostType` have an associated `User`?

**Reply by Lappihuan** [reply]
`#[Type(class: Post::class)]` or `#[Type(name: "Post" )]`
